### PR TITLE
Move eap test to boiler.mk scheme

### PR DESCRIPTION
**Pull request by jpereira**
Pretty simple, less code and reuse the same mechanism
to start/stop the radiusd service.

### CODE REVIEW

File: src/tests/eapol_test/all.mk [hunk 1]
```diff
@@ -1,44 +1,35 @@
...
 #
+#	Tests for EAP support
+#
+TEST := test.eap
+$(eval $(call TEST_BOOTSTRAP))
+
 TEST_PATH := ${top_srcdir}/src/tests/eapol_test
 CONFIG_PATH := $(TEST_PATH)/config
-
 OUTPUT := $(BUILD_DIR)/tests/eapol_test
```
Comment (alandekok) [inline src/tests/eapol_test/all.mk]: the `TEST_BOOTSTRAP` function defines `OUTPUT` already?

File: src/tests/eapol_test/all.mk [hunk 2]
```diff
@@ -1,44 +1,35 @@
...
-EAP_TYPES	:= $(patsubst rlm_eap_%.la,%,$(EAP_TARGETS))
-
+#
+#   This ensures that FreeRADIUS uses modules from the build directory
+#
+EAP_TARGETS      := $(filter rlm_eap_%,$(ALL_TGTS))
+EAP_TYPES        := $(patsubst rlm_eap_%.la,%,$(EAP_TARGETS))
 EAPOL_TEST_FILES := $(foreach x,$(EAP_TYPES),$(wildcard $(DIR)/$(x)*.conf))
-EAPOL_OK_FILES	 := $(patsubst $(DIR)/%.conf,$(OUTPUT)/%.ok,$(EAPOL_TEST_FILES))
+EAPOL_OK_FILES   := $(patsubst $(DIR)/%.conf,$(OUTPUT)/%.ok,$(EAPOL_TEST_FILES))
```
Comment (alandekok) [inline src/tests/eapol_test/all.mk]: There's no need to change whitespace here.

File: src/tests/eapol_test/all.mk [hunk 3]
```diff
@@ -175,7 +107,6 @@ ifneq (,$(findstring test,$(MAKECMDGOALS)))
 $(shell touch "$(OUTPUT)/eapol_test.skip")
 endif
 
-test.eap: $(OUTPUT)
-	${Q}echo "Skipping EAP tests due to previous build error"
+$(TEST): $(OUTPUT)
```
Comment (alandekok) [inline src/tests/eapol_test/all.mk]: I don't think this is needed any more with the `TEST_BOOTSTRAP` function

**Comment by arr2036** [comment]
If this is merged whilst the EAP tests are non-functional, i'm going to forcefully expunge it from the commit tree

**Comment by jpereira** [comment]
@arr2036 the `test.eap` is currently disabled as can be seen in https://github.com/FreeRADIUS/freeradius-server/commit/fc015119d90d35b913dbbdc65a05ff5038497e5c and btw, that changes keep the same behavior.

e.g:

1. Removing the broken tests.

```
[jpereira@sugarloaf freeradius-server.git]$ ../bypass-eapol.sh
########## By pass the eapol_test broken files!!
src/tests/eapol_test/peap-client-mschapv2.conf
src/tests/eapol_test/peap-eap-gtc.conf
src/tests/eapol_test/peap-mschapv2.conf
src/tests/eapol_test/ttls-chap.conf
src/tests/eapol_test/ttls-client-eap-mschapv2.conf
src/tests/eapol_test/ttls-client-eap-tls.conf
src/tests/eapol_test/ttls-eap-gtc.conf
src/tests/eapol_test/ttls-eap-mschapv2.conf
src/tests/eapol_test/ttls-mschapv2.conf
src/tests/eapol_test/ttls-pap.conf
[jpereira@sugarloaf freeradius-server.git]$
```

2. Running the test.

```
[jpereira@sugarloaf freeradius-server.git]$ make clean.test.eap test.eap
Clean up build/tests/eapol_test/radiusd.pid
rsa/server.pem: OK
ecc/server.pem: OK
rsa/ocsp.pem: OK
ecc/ocsp.pem: OK
Starting RADIUSD test server for (target=test.eap,port=12340,config_dir=src/tests/eapol_test/config,config_name=servers)
Executing: /bin/sh -c build/bin/local/radiusd -Pxxxl build/tests/eapol_test/radiusd.log -d src/tests/eapol_test/config -n servers -D share/dictionary/
ok
EAPOL_TEST aka-prime
EAPOL_TEST aka
EAPOL_TEST gtc
EAPOL_TEST md5
EAPOL_TEST mschapv2
EAPOL_TEST pwd
EAPOL_TEST sim
EAPOL_TEST tls
gmake[1]: Entering directory '/Volumes/Users/jpereira/Devel/FreeRADIUS/freeradius-server.git'
Clean up build/tests/eapol_test/radiusd.pid
gmake[1]: Leaving directory '/Volumes/Users/jpereira/Devel/FreeRADIUS/freeradius-server.git'
[jpereira@sugarloaf freeradius-server.git]$
```

**Comment by arr2036** [comment]
In addition to using the unified framework this PR needs to:
- Re-enable the EAP tests for Travis.
- Disable EAP virtual servers and modules that are there to support EAP methods which are disabled when we don't have OpenSSL.
- Disable EAP-TTLS and EAP-PEAP tests

**Comment by arr2036** [comment]
and @jpereira in future, when you're touching areas of the code or tests which are already failing in travis, at a bare minimum there needs to be discussion in the freeradius-devel channel, as it can make issues significantly harder to solve.

**Comment by jpereira** [comment]
@arr2036 @alandekok I added the capability to handle `IGNORED_LIST`

e.g:

```
[root@dev01 freeradius-server.git]# make clean.test.eap test.eap
Clean up build/tests/eapol_test/radiusd.pid
make[1]: Entering directory '/root/FreeRADIUS/freeradius-server.git/raddb/certs'
rsa/server.pem: OK
ecc/server.pem: OK
rsa/ocsp.pem: OK
ecc/ocsp.pem: OK
make[1]: Leaving directory '/root/FreeRADIUS/freeradius-server.git/raddb/certs'
EAPOL_TEST peap (Skipping, reenable by removing <peap> from 'IGNORED_EAP_TYPES' in src/tests/eapol_test/all.mk)
EAPOL_TEST ttls (Skipping, reenable by removing <ttls> from 'IGNORED_EAP_TYPES' in src/tests/eapol_test/all.mk)
EAPOL_TEST aka_prime (Skipping, reenable by removing <aka_prime> from 'IGNORED_EAP_TYPES' in src/tests/eapol_test/all.mk)
EAPOL_TEST aka (Skipping, reenable by removing <aka> from 'IGNORED_EAP_TYPES' in src/tests/eapol_test/all.mk)
Starting RADIUSD test server for (target=test.eap,port=12340,config_dir=src/tests/eapol_test/config,config_name=servers)
Executing: /bin/sh -c build/bin/local/radiusd -Pxxx -d src/tests/eapol_test/config -n servers -D share/dictionary/ -l build/tests/eapol_test/radiusd.log
ok
EAPOL_TEST gtc
EAPOL_TEST sim
EAPOL_TEST tls
EAPOL_TEST mschapv2
EAPOL_TEST pwd
EAPOL_TEST md5
make[1]: Entering directory '/root/FreeRADIUS/freeradius-server.git'
Clean up build/tests/eapol_test/radiusd.pid
make[1]: Leaving directory '/root/FreeRADIUS/freeradius-server.git'
[root@dev01 freeradius-server.git]#
```

**Comment by arr2036** [comment]
Great. Thanks!

Eap-aka prime and sim should be enabled thougb.

-Arran

**Comment by jpereira** [comment]
@arr2036 done.

**Comment by arr2036** [comment]
@alandekok this wasn't ready for merging, as evidenced by the build failures. @jpereira hasn't don't the mods-available and sites-available dependency checks to make it work with the non-openssl build

**Comment by alandekok** [comment]
I’ll take a look at that. IIRC those checks were missing before this PR was submitted.

**Comment by arr2036** [comment]
> In addition to using the unified framework this PR needs to:

> Re-enable the EAP tests for Travis.
> Disable EAP virtual servers and modules that are there to support EAP methods which are disabled when we don't have OpenSSL.
> Disable EAP-TTLS and EAP-PEAP tests

**Comment by arr2036** [comment]
I'm aware, which is why i *EXPLICITLY* included them in the list of conditions for merging this PR

**Pull request merged**